src/framework/theme/components/select/option-group.component.ts

```diff
@@ -26,13 +40,49 @@ export class NbOptionGroupComponent {
   }
   set disabled(value: boolean) {
     this._disabled = convertToBoolProperty(value);
+
+    if (this.options) {
+      this.updateOptionsDisabledState();
+    }
   }
   protected _disabled: boolean = false;
 
   @HostBinding('attr.disabled')
   get disabledAttribute(): '' | null {
     return this.disabled ? '' : null;
   }
+
+  @ContentChildren(NbOptionComponent, { descendants: true }) options: QueryList<NbOptionComponent<any>>;
+
+  ngAfterContentInit() {
+    if (this.options.length) {
+      this.asyncUpdateOptionsDisabledState();
+    }
+
+    this.options.changes
+      .pipe(takeWhile(() => this.alive))
+      .subscribe(() => this.asyncUpdateOptionsDisabledState());
+  }
+
+  ngOnDestroy() {
+    this.alive = false;
+  }
+
+  /**
+   * Sets disabled state for each option to current group disabled state.
+   */
+  protected updateOptionsDisabledState(): void {
+    this.options.forEach((option: NbOptionComponent<any>) => option.setDisabledByGroupState(this.disabled));
```

Comment:
Are you sure we have to use `NbOptionComponent<any>` here? Why can't we make `NbOptionGroupComponent` also generic?

Reply:
I like the idea, but then everyone who used NbOptionGroupComponent has to change their code. Not sure is it worth it since group has nothing to do with option values.